hostfactory/host_provider/src/cluster.py

```diff
@@ -43,108 +41,49 @@ def status(self):
-        exceeded_nodes = list(set(alloc_result.nodes) - set(filtered_nodes))
-        if len(exceeded_nodes) > 0:
-            self.logger.warning("In the allocation result %s nodes exceeded %s", len(exceeded_nodes), str(exceeded_nodes))
-            for node in self.node_mgr.get_nodes():
-                self.logger.debug("Node name %s Node state %s Node targetstate %s", node.name, node.state, node.target_state)
+        # Time in seconds to check waiting period after last capacity failure
+        capacity_failure_backoff = self.provider_config.get("capacity-failure-backoff", 300)
+        result = self.node_mgr.allocate({"weight": 1, "template_id": template_id, "capacity-failure-backoff": capacity_failure_backoff},
+                                slot_count=request['sets'][0]['count'],
+                                allow_existing=False)
+        self.logger.debug("Result of allocation %s", result)
         
-        self.logger.debug("Request id in add nodes %s",request['requestId'])    
-        request_id_start = f"{request['requestId']}-start"
-        request_id_create = f"{request['requestId']}-create"
-
-        bootup_resp = []
-        if len(filtered_nodes) > 0:
-            bootup_resp = self.node_mgr.bootup(nodes=filtered_nodes,
-            request_id_start=request_id_start, request_id_create=request_id_create
-            )
-        if (bootup_resp is None or []) or (bootup_resp.nodes is None or []):
-            return False
-        self.logger.debug("node bootup %s",bootup_resp)
-        self.logger.debug("node bootup requestids %s",bootup_resp.request_ids)
-        return (bootup_resp)
-    
-    def add_nodes(self, request, max_count):
-        def get_avail_count(status):
-            request_set = request['sets'][0]
-            machine_type = request_set["definition"]["machineType"]
-            nodearray_name = request_set['nodearray']
-            filtered = [x for x in status["nodearrays"] if x["name"] == nodearray_name]
-            if len(filtered) < 1:
-                raise RuntimeError(f"Nodearray {nodearray_name} does not exist or has been removed")
-            nodearray = filtered[0]
-            filtered_buckets = [x for x in nodearray["buckets"] if x["definition"]["machineType"] == machine_type]
-            if len(filtered_buckets) < 1:
-                raise RuntimeError(f"VM Size {machine_type} does not exist or has been removed from nodearray {nodearray_name}")
-            bucket = filtered_buckets[0]
-            return bucket["availableCount"]
+        if dry_run:
+            by_vm_size = partition(result.nodes, lambda node: node.vm_size)
+            for key,value in by_vm_size.items():
+                self.logger.debug("VM Size %s count %s", key, len(value))
+                print("Allocation result:")
+                print (key, len(value))
+            return True
+        if not dry_run and result:
```

Comment:
This can just be `if result`m since we always return in the prior. We want to make it clear that the result being successful is the more important part here, the check on not dry_run is not needed and distracts.